modules/api/src/main/java/org/apache/ignite/tx/TransactionOptions.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.tx;
+
+/**
+ * Ignite transaction options.
+ */
+public class TransactionOptions {
+    /** Transaction timeout. */
+    private long timeoutMillis = 0;
+
+    /** Read-only transaction. */
+    private boolean readOnly = false;
+
+    /**
+     * Returns transaction timeout, in milliseconds.
+     *
+     * @return Transaction timeout, in milliseconds.
+     */
+    public long timeoutMillis() {
+        return timeoutMillis;
+    }
+
+    /**
+     * Sets transaction timeout, in milliseconds.
+     *
+     * @param timeoutMillis Transaction timeout, in milliseconds.
+     * @return {@code this} for chaining.
+     */
+    public TransactionOptions timeoutMillis(long timeoutMillis) {
+        if (readOnly && timeoutMillis != 0) {
+            throw illegalStateException();
```

Comment:
Seems that I've missed that in initial discussion. Why do we consider timeout is an inappropriate attribute for readOnly transaction?

Reply:
@ascherbakoff can you please comment on this?

Reply:
Because it doesn't acquire any locks and can't be blocked in any way. But I recall the JDBC has query timeout, and possibly I'm wrong here. This timeout can be used to cancel misbehaving query from a client side, same as Ignite2 does.

Reply:
I'm not sure if JDBC and `IgniteTransactions` interface can work together.

In any case, transaction timeouts are not implemented yet, should we postpone this decision and remove the exception?

Reply:
Yep, I'd remove the exception.

Reply:
Done